src/pages/instances/actions/snapshots/ConfigureSnapshotModal.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { FC, KeyboardEvent } from "react";
+import { Button, Modal } from "@canonical/react-components";
+import { LxdInstance } from "types/instance";
+import { useFormik } from "formik";
+import { updateInstance } from "api/instances";
+import { queryKeys } from "util/queryKeys";
+import { EditInstanceFormValues } from "pages/instances/EditInstanceForm";
+import * as Yup from "yup";
+import { useQueryClient } from "@tanstack/react-query";
+import { useNotify } from "context/notify";
+import SnapshotsForm from "pages/instances/forms/SnapshotsForm";
+import { useParams } from "react-router-dom";
+import { getInstanceEdit, getInstancePayload } from "util/instanceEdit";
+
+interface Props {
+  instance: LxdInstance;
+  close: () => void;
+}
+
+const ConfigureSnapshotModal: FC<Props> = ({ instance, close }) => {
+  const notify = useNotify();
+  const { project } = useParams<{ project: string }>();
+  const queryClient = useQueryClient();
+
+  const InstanceSchema = Yup.object().shape({
+    name: Yup.string().required("Instance name is required"),
+    instanceType: Yup.string().required("Instance type is required"),
+  });
```

Comment:
For creation it is different, but we can reuse for the two edits.

Reply:
The only difference that I see is that the name is optional in creation and required in edit. But I would try to find a workaround for this, and still reuse the same schema for both creation and edit. Especially because we aren't doing the "deduplicate" test on edit, which should be done instead.
You can make it a function `getInstanceSchema` and pass the mutation (`optional` / `required`) to it maybe?

Reply:
I think it is better not to mix creation/editing schemas.